packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts

```diff
@@ -66,15 +80,53 @@ export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
     });
     return quantities;
   }
+
+  public async execute(response: any): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = await new CliCommandExecutor(this.build(response), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    const result = await new CommandOutput().getCmdResult(execution);
+    const resultJson = JSON.parse(result);
+    if (resultJson.status === 0 && this.openAfterRetrieve) {
+      const extensions = MetadataDictionary.getInfo(
+        resultJson.result.inboundFiles[0].type
+      )?.extensions;
+
+      resultJson.result.inboundFiles.map( async (item: any) => {
```

Comment:
teenie nit, probably can use `for (const file of inboundFiles)` or `forEach` here since we don't need to build a new array.

Reply:
Done!